File: javascript/bob/bob.js
var Bob = function() {
	this.hey = function(input) {
		isQuestion = function(input) {
			return input.charAt(input.length - 1) === '?';
		}
		isYelling = function(input) {
			return input.toUpperCase() === input && /[A-Z]/.test(input);
		}
		isSilent = function(input) {
			return input === '' || /^\s+$/.test(input);
		}
		if (isYelling(input)) {
			return 'Whoa, chill out!';
		}
		if (isQuestion(input)) {
			return 'Sure.';
		}
		if (isSilent(input)) {
			return 'Fine. Be that way!';
		}
		return "Whatever.";
	};
};
module.exports = Bob;

File: python/bob/bob.py
"""'Bob' from exercism.io"""

import re


def hey(phrase):
    """Core bob function... issues a reply to 'phrase'."""

    phrase = phrase.strip()
    question = re.compile(r'.*\?$')
    letters = ''.join([l for l in phrase if re.match(r'\w', l)])
    yelling = False

    if (letters.isupper() is True):
        yelling = True

    if (yelling is True):
        if (question.match(phrase)):
            return "Calm down, I know what I'm doing!"
        else:
            return 'Whoa, chill out!'

    if (question.match(phrase)):
        return 'Sure.'

    if (letters.strip() == ''):
        return 'Fine. Be that way!'

    return 'Whatever.'
